share/spack/gitlab/cloud_pipelines/stacks/aws-pcluster-icelake/spack.yaml

```diff
@@ -0,0 +1,58 @@
+spack:
+  view: false
+
+  definitions:
+  - compiler_specs:
+    - gcc
+    - gettext
+
+  - compiler_target:
+    - '%gcc@7.3.1 target=x86_64_v3'
+
+  - optimized_configs:
+    # - gromacs
+    - lammps
+    # - mpas-model
+    - openfoam
+    # - palace
+    # - py-devito
+    # - quantum-espresso
+    # - wrf
+
+  - optimized_libs:
+    - mpich
+    - openmpi
+
+  specs:
+  - matrix:
+    - - $compiler_specs
+    - - $compiler_target
+  - $optimized_configs
+  # - $optimized_libs
+
+  mirrors: { "mirror": "s3://spack-binaries/develop/aws-pcluster" }
+
+  ci:
+    pipeline-gen:
+    - build-job:
+        image: { "name": "ghcr.io/spack/pcluster-amazonlinux-2:latest", "entrypoint": [""] }
+        before_script:
+        - - . "./share/spack/setup-env.sh"
+          - . /etc/profile.d/modules.sh
+          - spack --version
+          - spack arch
+        # Setup postinstall Spack as upstream installation
+        - - if [[ -f /bootstrap/spack/etc/spack/packages.yaml ]]; then cp /bootstrap/spack/etc/spack/packages.yaml ./etc/spack/; fi
+          - if [[ -f /bootstrap/spack/etc/spack/config.yaml ]]; then cp /bootstrap/spack/etc/spack/config.yaml ./etc/spack/; fi
+          - if [[ -f /bootstrap/spack/etc/spack/modules.yaml ]]; then cp /bootstrap/spack/etc/spack/modules.yaml ./etc/spack/; fi
+          - if [[ -f /bootstrap/spack/etc/spack/mirrors.yaml ]]; then cp /bootstrap/spack/etc/spack/mirrors.yaml ./etc/spack/; fi
+          - if [[ -d /bootstrap/spack/opt/spack ]]; then spack config add "upstreams:postinstall:install_tree:/bootstrap/spack/opt/spack"; fi
+        - - /bin/bash "${SPACK_ARTIFACTS_ROOT}/postinstall.sh" -fg
+          - spack config --scope site add "packages:all:target:\"target=${SPACK_TARGET_ARCH}\""
+    - signing-job:
+        before_script:
+          # Do not distribute Intel & ARM binaries
+          - - for i in $(aws s3 ls --recursive ${SPACK_REMOTE_MIRROR_OVERRIDE}/build_cache/ | grep intel-oneapi | awk '{print $4}' | sed -e 's?^.*build_cache/??g'); do aws s3 rm ${SPACK_REMOTE_MIRROR_OVERRIDE}/build_cache/$i; done
+            - for i in $(aws s3 ls --recursive ${SPACK_REMOTE_MIRROR_OVERRIDE}/build_cache/ | grep armpl | awk '{print $4}' | sed -e 's?^.*build_cache/??g'); do aws s3 rm ${SPACK_REMOTE_MIRROR_OVERRIDE}/build_cache/$i; done
```

Comment:
Do I understand this correctly?  You are deleting from the mirror some (possibly large) subset of the binaries just built, in order to avoid re-distribution?  I thought it was going to be ok to leave these in the stack-specific mirrors, as long as we don't included them at the root.  The way it is here, isn't it going to force every matching spec to be rebuilt from source on every pipeline?

Reply:
No, I am only deleting software that's distributed as binaries from Intel/ARM anyway. This is only oneapi and armPL software. On the one hand we are on the save side for re-distribution and on the hand the installation "from source" is downloading pre-built binaries anyway.

Reply:
What I should have said here, but I just didn't think of at the time, is that some time ago we made it an error (see [here](https://github.com/spack/spack/blob/60b4e2128ba00798a6d5c4d0bb1e44c4230ebde6/lib/spack/spack/cmd/ci.py#L538)) for dependencies in a pipeline rebuild job to be installed from source.  The definition of "from source" there doesn't consider that it might be downloading pre-built binaries, it's either from a spack buildcache or it's "from source".